.github/workflows/Check.yml

```diff
@@ -0,0 +1,38 @@
+name: Code checks
+
+on:
+  pull_request:
+  push:
+    branches: ["master"]
+
+jobs:
+
+  pre-commit:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+      - uses: pre-commit/action@2c7b3805fd2a0fd8c1884dcaebf91fc102a13ecd # v3.0.1
```

Comment:
Why is there a specific commit here and not `@v3.0.1` ?

Reply:
https://docs.github.com/en/actions/security-guides/security-hardening-for-github-actions#using-third-party-actions